jgrapht-core/src/main/java/org/jgrapht/GraphMetrics.java

```diff
@@ -111,7 +110,7 @@
                     return 1;
 
         if(graph.getType().isUndirected()) {
-            NeighborIndex<V,E> neighborIndex=new NeighborIndex<>(graph);
+            NeighborCache<V, E> neighborIndex=new NeighborCache<>(graph);
```

Comment:
Now we can pull this out above the if since we use the same neighborIndex in both branches.